torch/csrc/jit/ir.cpp

```diff
@@ -1686,6 +1686,37 @@ Value* Graph::insertUncheckedCast(Value* v, TypePtr type) {
   return n->output();
 }
 
+Value* Graph::insertToList(Value* v, TypePtr type) {
+  int dim = 0;
+  TypePtr ptr = type;
+
+  // Unwrap the type to determine the number of dimensions.
+  while (auto list_type = ptr->cast<ListType>()) {
+    ptr = list_type->getElementType();
+    ++dim;
+  }
+
+  // Encode the base element type as an integer.
+  int elem_ty = 0;
+  if (ptr == IntType::get()) {
+    elem_ty = 0;
+  } else if (ptr == FloatType::get()) {
+    elem_ty = 1;
+  } else if (ptr == BoolType::get()) {
+    elem_ty = 2;
+  } else {
+    TORCH_INTERNAL_ASSERT(false, "Unsupported data type for tolist");
+  }
```

Comment:
I don't like this either, but I have to do this to avoid capturing the output type in the operator lambda in the interpreter.